### ISSUE TITLE
Super Mode: assert_armored throws seg fault

### ISSUE THREAD

**Issue by NoahCardoza** [dashingsoft/pyarmor#448]
Not sure what info I can provide except I get a seg fault whenever I call `assert_armored` in Super Mode.

I haven't tested it out side of super mode. I thought it could be that I was using restrict mode 4 on some of the files, so I reverted to restrict mode 1 for all my files and it didn't make a difference.

UPDATE:

Just tested in Windows and got:

![image](https://user-images.githubusercontent.com/10343470/105252150-c6b1d380-5b31-11eb-813a-17d1435aee9c.png)

UPDATE:

Just tested without advanced mode and got:

```
Traceback (most recent call last):
  File "dist/TestIAO/example.py", line 1, in <module>
    import harvester
  File "</Users/vulcan/projects/CaptchaHarvester/server/factory/dist/TestIAO/harvester/__init__.py>", line 3, in <module>
  File "<frozen src>", line 4, in <module>
  File "/Users/vulcan/projects/CaptchaHarvester/server/factory/dist/TestIAO/harvester/pytransform/__init__.py", line 168, in wrap_execute
    dlfunc(names)
RuntimeError: Protection Fault: this function is not pyarmored
```

I double checked and I can confirm that the function is indeed pyarmored.

**Comment by NoahCardoza** [comment]
All right, I learned a couple of interesting things about `assert_armored`.

1. `assert_armored` doesn't like classes, so I had to replace those checks with `ClassName.__init__`.
2. `assert_armored` doesn't like checking methods that already have assert_armored decorators.

It would be cool if there were examples on how to handle these cases. It took me a couple hours to figure this out through trial and error since Segfaults aren't very informative.

I knew it had to me using  `assert_armored` wrong 😂
